Followup suggestions from ADR-027 updates (#7232) (#7379)
cosmos/cosmos-sdk: docs/architecture/adr-027-deterministic-protobuf-serialization.md

## Abstract

Fully deterministic structure serialization, which works across many languages and clients,

> is needed for the use case of structure signatures. We need to be sure that whenever we serialize

Reviewer: What are `structure signatures`. That's really not very clear

Reviewer 2: Maybe `is needed for generating a signature payload.` ?

Reviewer 3: How about _is needed to sign a structure._?

Reviewer 2: My reasoning was, `Fully deterministic structure serialization is needed...` to sign a struct, an array, a string, an integer, any kind of signature payload.

Reviewer: Message is the protobuf term.

Author: Updated to "needed when signing messages".  I hope that is more clear.

> is needed for the use case of structure signatures. We need to be sure that whenever we serialize

Author: ```suggestion
is needed when signing messages. We need to be sure that whenever we serialize
```
